packages/openapi-2-kong/src/declarative-config/plugins.ts

```diff
@@ -36,46 +39,88 @@ const generatePlugin = (tags: string[]) => ([key, value]: Entry<PluginItem>): DC
  * See: https://github.com/Kong/kong-plugin-enterprise-request-validator/pull/34/files#diff-1a1d2d5ce801cc1cfb2aa91ae15686d81ef900af1dbef00f004677bc727bfd3cR284
  */
 export const ALLOW_ALL_SCHEMA = '{}';
-
+const $schema = 'http://json-schema.org/schema#';
 const DEFAULT_PARAM_STYLE = {
   header: 'simple',
   cookie: 'form',
   query: 'form',
   path: 'simple',
 };
 
-const generateParameterSchema = (operation?: OA3Operation) => {
+const resolveParameter = ($refs: SwaggerParser.$Refs, parameter: OpenAPIV3.ParameterObject | OpenAPIV3.ReferenceObject): OpenAPIV3.ParameterObject => {
+  if ('$ref' in parameter) {
+    const components = $refs.get('#/components');
+    const dereferenced = $refs.get(parameter.$ref);
+    const param = omit(parameter, ['$ref']);
```

Comment:
We could remove the omit dep here:

```suggestion
const { $ref, ...param } = parameter;
```